pkg/controller/migplan/validation.go

```diff
@@ -474,6 +481,40 @@ func (r ReconcileMigPlan) validateDestinationCluster(plan *migapi.MigPlan) error
 	return nil
 }
 
+func (r ReconcileMigPlan) validateOperatorVersions(plan *migapi.MigPlan) error {
+	destRef := plan.Spec.DestMigClusterRef
+	srcRef := plan.Spec.SrcMigClusterRef
+
+	srcCluster, err := migapi.GetCluster(r, srcRef)
+	if err != nil {
+		return liberr.Wrap(err)
+	}
+
+	destCluster, err := migapi.GetCluster(r, destRef)
+	if err != nil {
+		return liberr.Wrap(err)
+	}
+
+	srcClusterOperatorVersion, err := srcCluster.GetOperatorVersion(r)
+	destClusterOperatorVersion, err := destCluster.GetOperatorVersion(r)
+	//Not sure how to access the host operator version from here?
```

Comment:
I think you could just read the MigCluster conditions here and avoid doing any remote API calls.

Since we already compare the version with host in the MigCluster validation, we could just do a lookup on MigCluster conditions with:

```
SomeMigCluster.HasCondition(OperatorVersionMismatch)
```